graphql/schemabuilder/reflect_test.go

```diff
@@ -210,20 +226,65 @@ func TestEnumMapWrongArg(t *testing.T) {
 
 }
 
-func TestEnumMapKeys(t *testing.T) {
+func TestEnumMapWrongArg2(t *testing.T) {
+
 	schema := NewSchema()
+
+	type enumType3 float64
+
 	defer func() {
 		r := recover()
 		if r == nil {
 			t.Error("expected a panic")
 		}
-		assert.Equal(t, "keys are not strings", r)
 	}()
-	type enumType int32
-	schema.Enum(enumType(1), map[int32]int32{
-		1: 1,
+
+	schema.Enum(enumType3(1), map[string]interface{}{
+		"firstField":  interface{}(1),
+		"secondField": interface{}(2),
+		"thirdField":  interface{}(3),
 	})
```

Comment:
i think there are ways we could support this case: https://play.golang.org/p/9bjfb8w_8h2

Reply:
It doesn't differentiate between int32 and int

Reply:
could you paste a code example? https://play.golang.org/p/ItVllolNiF2

Reply:
This particular test case will still fail because the integers aren't of int32 type

Reply:
ah, that makes sense.